Consul/Interfaces/IAgentEndpoint.cs

```diff
@@ -44,7 +45,7 @@ public interface IAgentEndpoint
         string NodeName { get; }
         Task<string> GetNodeName(CancellationToken ct = default(CancellationToken));
         Task PassTTL(string checkID, string note, CancellationToken ct = default(CancellationToken));
-        Task<QueryResult<Dictionary<string, Dictionary<string, dynamic>>>> Self(CancellationToken ct = default(CancellationToken));
+        Task<QueryResult<Dictionary<string, Dictionary<string, JsonElement>>>> Self(CancellationToken ct = default(CancellationToken));
```

Comment:
We need to be really careful with changing the public API to avoid runtime failures when [diamond dependency issue](https://codeblog.jonskeet.uk/2019/06/30/versioning-limitations-in-net/) occurs. 
I think that in this case the best would be to keep using the `dynamic` "type". To make it compile we could simply add an explicit package reference to `Microsoft.CSharp Version="4.3.0"`. It will not by any real dependency addition as this package is currently referenced transitively via `Newtonsoft.Json`.